extras/linter_install_tutorial.md

```diff
@@ -0,0 +1,127 @@
+7. Then call command ``pre-commit install`` to install the pipeline to run after the ``git commit`` command is called.
+8. Now the pipeline is installed, and you can test it by calling ``pre-commit``.
+9. Now you can proceed with normal code development and GitHub processes
+10. When you call `git commit`, you will see a few checks being at the start of the committing process.
+11. Checkpoints will fail as files are modified or flake8 errors persist
+12. Review that changes files are to one's liking or manually make changes
+13. Lastly, call ``git add`` on the changed files then call the same ``git commit`` command again.
+
+## Warnings
+
+This pipeline is designed to run black on jupyter notebooks and does not run on individual python scripts.
+The flake8 step will complain about non-formatted python scripts as no changes would have been made at that point.
+Simply run the command `black <python script>` and re-add the changed file.
+
+## Configuration Files 
+
+### Pre-commit
+
+```yaml
+repos:
+
+- repo: https://github.com/mwouts/jupytext
+  rev: 41b7e9c93b50da1d6feb486aceb2c4d534374090
+  hooks:
+  - id: jupytext
+    name: jupytext_auto_linter
+    args: [--set-formats, 'ipynb,py', --from, ipynb, --pipe, black, --to, py:light, --sync]
+
+- repo: https://gitlab.com/pycqa/flake8
+  rev: 6de8252c035844f1e679f509b5f37340b44d5c39
+  hooks:
+  - id: flake8
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: 43f5ffaeab549e15168317892bf217dd450aa41b
+    hooks:
+    - id: trailing-whitespace
+```
+
+#### Pre-commit with Specified Folders
+
+If one wants to allocate all scripts to one folder and notebooks to another folder use this file instead.
+Just replace notebooks and scripts with the filenames you need. [Documentation here](https://github.com/mwouts/jupytext/blob/master/docs/config.md#per-notebook-configuration)
+
+```yaml
+repos:
+
+- repo: https://github.com/mwouts/jupytext
+  rev: 41b7e9c93b50da1d6feb486aceb2c4d534374090
+  hooks:
+  - id: jupytext
+    name: jupytext_auto_linter
+    args: [--set-formats, 'notebook_folder_name//ipynb,script_folder_name//py', --from, ipynb, --pipe, black, --to, py:light, --sync]
+
+- repo: https://gitlab.com/pycqa/flake8
+  rev: 6de8252c035844f1e679f509b5f37340b44d5c39
+  hooks:
+  - id: flake8
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: 43f5ffaeab549e15168317892bf217dd450aa41b
+    hooks:
+    - id: trailing-whitespace
+```
+
+Depending on repository organization some may want to have files be in specific folders rather than in a centralized way.
+Currently, there isn't a great way to accomplish this feature without manually calling ``jupytext --set-formats 'notebook_folder_name//ipynb,script_folder_name//py'``.
+Then, remove the ``--set-formats, 'notebook_folder_name//ipynb,script_folder_name//py'`` part from the config file and run pre-commit like normal.
```

Comment:
Interesting, didn't realize you'd need to re-run the pre-commit again afterward. Thanks for adding this option though it sounds like it might not be worth it give the manual step possibly

Reply:
Ah you don't have to run pre-commit. I just incorporated it this way to have the user test the change in case anything goes wrong.

Reply:
I'll test this feature next week and see what I find. Thanks for adding this!

Reply:
Getting the following error:
```
(generic_expression) alexandra@alexandra:~/Documents/Repos/generic-expression-patterns$ jupytext --set-formats 'LV_analysis//ipynb,LV_analysis//nbconverted//py'
Traceback (most recent call last):
  File "/home/alexandra/anaconda3/envs/generic_expression/bin/jupytext", line 10, in <module>
    sys.exit(jupytext())
  File "/home/alexandra/anaconda3/envs/generic_expression/lib/python3.7/site-packages/jupytext/cli.py", line 402, in jupytext
    exit_code += jupytext_single_file(nb_file, args, log)
  File "/home/alexandra/anaconda3/envs/generic_expression/lib/python3.7/site-packages/jupytext/cli.py", line 418, in jupytext_single_file
    raise ValueError(msg)
ValueError: Missing notebook path.
```

Reply:
Ok. Did you manually call this? If so you need to add the file path as the next argument. Also remove the double forward slash between LV_analysis and nbconverted. Working code below:

```bash
jupytext --set-formats 'LV_analysis//ipynb,LV_analysis/nbconverted//py' LV_analysis/notebook name.ipynb
```

Reply:
Yup this works!
`jupytext --set-formats 'LV_analysis//ipynb,LV_analysis/nbconverted//py' LV_analysis/*.ipynb`

That's a shame it can't be automated but at least it works. Thank you for all your help!